shm_wrapper/shm_wrapper.c

```diff
@@ -207,16 +207,16 @@ void print_dev_ids ()
 {
 	dev_id_t dev_ids[MAX_DEVICES];
 	uint32_t catalog;
-	
+
 	get_device_identifiers(dev_ids);
 	get_catalog(&catalog);
-	
+
 	if (catalog == 0) {
 		printf("no connected devices\n");
 	} else {
 		for (int i = 0; i < MAX_DEVICES; i++) {
 			if (catalog & (1 << i)) {
-				printf("dev_ix = %d: type = %d, year = %d, uid = %llu\n", i, dev_ids[i].type, dev_ids[i].year, dev_ids[i].uid);
+				printf("dev_ix = %d: type = %d, year = %d, uid = %lu\n", i, dev_ids[i].type, dev_ids[i].year, dev_ids[i].uid);
```

Comment:
uint64 is a long long

Reply:
Ah I was getting a warning for having %llu instead of %lu

Reply:
oh probably becuase you're on ubuntu?? C sucks so long has different meanings on each system